### PR TITLE
enhance helm script

### PR DESCRIPTION
**Pull request by jeffvance**
Signed-off-by: jeffvance <jeff.h.vance@gmail.com>

A couple of enhancements to the helm.sh script.
. `clean` works even if helm was temporarily installed
. added `reset` to separate helm reset and k8s cleanup from rm'ing the temp helm dir

cc @travisn

### CODE REVIEW

File: tests/scripts/helm.sh [hunk 1]
```diff
@@ -26,19 +24,29 @@ detectArch() {
...
 
-install() {
-    if ! helm_loc="$(type -p "helm")" || [[ -z ${helm_loc} ]]; then
-        # Download and unpack helm
-        local dist
-        dist="$(uname -s)"
+# Set the global HELM variable to the helm command. Install helm to a temp 
+# location if it's not present.
+install_helm() {
+    HELM="helm"
```
Comment (galexrt) [inline tests/scripts/helm.sh]: I would be for moving the `HELM` var outside this function as was previously.

Comment (jeffvance) [reply tests/scripts/helm.sh]: ok. I agree it's cleaner.

File: tests/scripts/helm.sh [hunk 2]
```diff
@@ -26,19 +24,31 @@ detectArch() {
...
+    elif ! which ${helm} &>/dev/null; then
+        echo "Installing helm..." >&2
         # shellcheck disable=SC2021
-        dist=$(echo "${dist}" | tr "[A-Z]" "[a-z]")
         mkdir -p "${temp}"
-        wget "https://storage.googleapis.com/kubernetes-helm/helm-v2.13.1-${dist}-${arch}.tar.gz" -O "${temp}/helm.tar.gz"
-        tar -C "${temp}" -zxvf "${temp}/helm.tar.gz"
-        HELM="${temp}/${dist}-${arch}/helm"
+	local helm_url="https://storage.googleapis.com/kubernetes-helm/helm-v2.13.1-${dist}-${arch}.tar.gz"
+	local helm_gz="${temp}/helm.tar.gz"
```
Comment (galexrt) [inline tests/scripts/helm.sh]: Same.

File: tests/scripts/helm.sh [hunk 3]
```diff
@@ -26,19 +24,31 @@ detectArch() {
...
+        helm="${tmp_helm}"
+    elif ! which ${helm} &>/dev/null; then
+        echo "Installing helm..." >&2
         # shellcheck disable=SC2021
-        dist=$(echo "${dist}" | tr "[A-Z]" "[a-z]")
         mkdir -p "${temp}"
-        wget "https://storage.googleapis.com/kubernetes-helm/helm-v2.13.1-${dist}-${arch}.tar.gz" -O "${temp}/helm.tar.gz"
-        tar -C "${temp}" -zxvf "${temp}/helm.tar.gz"
-        HELM="${temp}/${dist}-${arch}/helm"
+	local helm_url="https://storage.googleapis.com/kubernetes-helm/helm-v2.13.1-${dist}-${arch}.tar.gz"
```
Comment (galexrt) [inline tests/scripts/helm.sh]: The indentation is wrong here. It has to be one further in.

File: tests/scripts/helm.sh [hunk 4]
```diff
@@ -26,19 +24,31 @@ detectArch() {
...
 
-install() {
-    if ! helm_loc="$(type -p "helm")" || [[ -z ${helm_loc} ]]; then
-        # Download and unpack helm
-        local dist
-        dist="$(uname -s)"
+# Echo the name or path of the helm command. Install helm to a temp location
+# if it's not present.
+install_helm() {
+    local helm="helm"
```
Comment (galexrt) [inline tests/scripts/helm.sh]: Why are you using a local variable instead of overriding the variable and later on calling the function?

Comment (jeffvance) [reply tests/scripts/helm.sh]: I understood an earlier comment to suggest not to define a global var in this func. I agree (to improve structure and readability) and so I now use a local var to hold the actual helm cmd, then return its value  (via stdout) and the main script assigns it to the HELM var.

File: tests/scripts/helm.sh (hunk 3 above)
Comment (jeffvance) [reply tests/scripts/helm.sh]: right, mistakenly used a tab here and below.

File: tests/scripts/helm.sh (hunk 4 above)
Comment (galexrt) [reply tests/scripts/helm.sh]: I see, my first comment was meant that the variable from line 6 is used and the function just overrides it when helm is not installed.

Comment (jeffvance) [reply tests/scripts/helm.sh]: HELM now is only assigned to in one place, so it's easy to know its value. Are you ok with this?

File: tests/scripts/helm.sh [hunk 5]
```diff
@@ -26,19 +24,31 @@ detectArch() {
...
+    local dist="$(uname -s)"
+    dist="$(echo "${dist}" | tr '[A-Z]' '[a-z]')"
+    local tmp_helm="${temp}/${dist}-${arch}/helm"
+
+    if [[ -f "${tmp_helm}" ]]; then # helm installed by this script
+        helm="${tmp_helm}"
+    elif ! which ${helm} &>/dev/null; then
+        echo "Installing helm..." >&2
         # shellcheck disable=SC2021
```
Comment (galexrt) [inline tests/scripts/helm.sh]: You probably need to remove the `shellcheck` comment now unless it is needed. Could you check that with shellcheck if this is still needed?

File: tests/scripts/helm.sh [hunk 6]
```diff
@@ -76,33 +86,49 @@ install() {
...
-
 }
 
+# Reset helm or reset & rm temp helm
 helm_reset() {
-    "${HELM}" reset
+    local do_rm="$1"
+
+    ${HELM} reset
     # shellcheck disable=SC2021
```
Comment (galexrt) [inline tests/scripts/helm.sh]: Same here, is the shellcheck disable comment still needed?

File: tests/scripts/helm.sh (hunk 5 above)
Comment (jeffvance) [reply tests/scripts/helm.sh]: Thanks for your reviews. I've removed this.

**Comment by jeffvance** [comment]
@galexrt I think I've covered your review comments (which, btw, I appreciate). If you are satisfied now please approve.

**Comment by galexrt** [comment]
@jeffvance Could you please go ahead and rebase on latest `master`, maybe that makes the CI issue go away. Thanks!

**Comment by jeffvance** [comment]
Yes I will right after our cephlacon session.
jeff

**Comment by galexrt** [comment]
@jeffvance Ping. :slightly_smiling_face:

**Comment by jeffvance** [comment]
@galexrt thanks for the reminder! It's rebased now.

**Comment by galexrt** [comment]
@jeffvance Your rebase caused a new commit?
Could you squash your commits into one and check the rebase on latest `master` again. Rebase shouldn't cause a new commit.

**Comment by galexrt** [comment]
@jeffvance Ping. Please rebase on latest `master` and check the CI if the change fails it after your push.

**Comment by jeffvance** [comment]
On vacation this week. Will get to it next week.

**Comment by leseb** [comment]
Closing this due to inactivity. Feel free to re-open. Thanks.

**Pull request closed without merging**